### DISCUSSION TITLE
Kubernetes install isn't working

### DISCUSSION THREAD

**Discussion by ea-p** [firefly-iii/firefly-iii#4523] [Firefly III] [answered]
**Bug description**
I am trying to install Firefly III with kubernetes with the kubernetes installer from https://github.com/ea-p/kubernetes, and my problem is:

I managed to get a reverse proxy with apache going to access the service, but the best result I can get is a 

SQLSTATE[HY000] [2002] No route to host (SQL: select * from information_schema.tables where table_schema = fireflyiii and table_name = migrations and table_type = 'BASE TABLE')

**Steps to reproduce**
Clone https://github.com/ea-p/kubernetes,  and run  -  kubectl apply -k ./ 
Cloning https://github.com/firefly-iii/kubernetes the pods don't even fire up. They keep pending because they can't pull a PVC

The same error appears if i /bin/bash into the firefly-iii pod and try to execute:
php artisan migrate:refresh --seed

I haven't redone it now, but I recall getting the same error if I try to do this with docker and a databse in docker so I think this might be some kind of networking issue.
 
[logs.log](https://github.com/firefly-iii/firefly-iii/files/6177986/logs.log)

**Extra info**
Screenshot:
![Untitled](https://user-images.githubusercontent.com/19634656/111925463-a65eb080-8ab1-11eb-9e55-4080d88913db.png)

OS: Centos 8.3

kubectl version output: 
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T14:30:33Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T14:23:04Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

**Bonus points**
- [x]  I searched and nobody reported this bug before
- [ ] I have added a stack trace from my log files  Going to activate debug level tomorrow, it's a tad bit more complicated with k8s.
- [x] I have added a screenshot.
- [ ] I was able to replicate it on the demo site https://demo.firefly-iii.org/ - Installation issue, don't think this can be replicated on the demo site.

**Comment by JC5** [comment]
I know nothing of kubernetes, I just provide the files. If somebody else can help, please do!

**Comment by ea-p** [comment]
Ok. Seems like google is giving better results after I submitted this.
The problem is two-fold.

1. firewalld was blocking traffic. To get rid of the 'No route to host' error just do `firewall-cmd --add-masquerade --permanent`
2. Afterwards you get an error that no `fireflyiii` database is available. The databse needs to be initialized.
  a. as root do `kubectl exec -it <db-pod-name> -- mysql -p` then enter root pass and do `CREATE DATABASE fireflyiii;`
  b. as root do `kubectl exec -it <firefly-iii-pod-name> -- /bin/bash` then do `php artisan migrate:refresh --seed`, `php artisan firefly-iii:upgrade-database` , `php artisan passport:install


The problem with this is that the cluster does not have high availability. If the firefly-iii pod gets restarted it stays in pending. I am no kubernetes expert either, but I believe the PVC doesn't get released. I'll keep trying, but if this doesn't work I'll try simple docker. Having to reinstalll every time a pod would fail or get restarted is not the purpouse of using k8s.

**Reply by creylopez** [reply]
<img width="630" alt="Captura de pantalla 2021-04-09 a las 8 18 45" src="https://user-images.githubusercontent.com/54448654/114137347-5767a680-990c-11eb-9367-ba65e2ae4125.png">
I have a similar problem but this time with the database access. I'm running Docker desktop with Kubernetes enabled on macOS. 

I've tried to shell into the mysql deployment and enter mysql with mysql -u root -p, but the password I put in kustomization.yaml seems not to work.

Thanks.

**Comment by devfaz** [comment, marked as answer]
I run firefly on k8s will update/provide my files asap.

Remind me if I forget to do it.

**Reply by creylopez** [reply]
Super! Thanks a lot

**Reply by devfaz** [reply]
https://github.com/firefly-iii/kubernetes/pull/6

**Reply by ea-p** [reply]
Thank you!
